[DONE] Add Functionality to Text Search Screen

liza-khr opened the pull request:
Changes:
- Functionality for TextSearchScreen
- Changes in TextLookupControl to allow adding items to cart
- GUI keyboard input

calderslom reviewed (approved):
Approved. Ready to add Text Search use case to main

RowanDavitt reviewed (changes requested):
- doing a text search doesn't clear the previous results, so the results pile up

- stuff typed with the gui keyboard is sort of carried across text field (for exampleif you type a password to log in to the attendant station, the password will be added to the item search textfield when you start typing)

- the textfield in item search is still editable

- apparently presentCashScreen is meant to have been deleted



- there are also some unimplemented method errors that should be fixed at some point

- the old hardware directory should be deleted at some point

liza-khr commented:
> * doing a text search doesn't clear the previous results, so the results pile up
Fixed
> * stuff typed with the gui keyboard is sort of carried across text field (for exampleif you type a password to log in to the attendant station, the password will be added to the item search textfield when you start typing)
I don't know how to fix this. Since we're low on time I think it's okay to leave it
> * the textfield in item search is still editable
Fixed
> * apparently presentCashScreen is meant to have been deleted
Done
> * there are also some unimplemented method errors that should be fixed at some point
Fixed
> * the old hardware directory should be deleted at some point
Done
